cmd/quadlet/main.go

```diff
@@ -210,6 +211,66 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+func isSha256Digest(imageName string) bool {
+	if strings.HasPrefix(imageName, "sha256:") {
+		return true
+	}
+
+	// only fully specified digests are matched
+	if len(imageName) != 64 {
+		return false
+	}
+
+	for _, c := range imageName {
+		if !unicode.Is(unicode.Hex_Digit, c) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isFullyQualifiedName(imageName string) bool {
```

Comment:
If the goal is to detect digests, `isUnambiguousName`?

Alternatively, if the goal is primarily to avoid the short name parsing overhead, even digested short names should be warned against, shouldn’t they?

Reply:
We verify that the length is 64 when doing, so we should warn on "short-identifier" (as named in the c/i/docker/reference/reference.go grammar comment). T